src/bitcoin/Transaction.mo

```diff
@@ -283,13 +301,23 @@ module {
 
       let sha_outputs : [Nat8] = Blob.toArray(Sha256.fromArray(#sha256, outputs_bytes));
 
-      // (ext_flag * 2) + annex_present
-      let spend_type : [Nat8] = [0x00];
-
       var input_index_buffer = Array.init<Nat8>(4, 0);
       Common.writeLE32(input_index_buffer, 0, txInputIndex);
       let input_index = Array.freeze(input_index_buffer);
 
+      // spend_type = (ext_flag * 2) + annex_present
+      let (spend_type, scriptpath_bytes) : ([Nat8], [Nat8]) = switch (maybe_leaf_hash) {
+        case (?leaf_hash) {
+          assert leaf_hash.size() == 32;
+          let KEY_VERSION_0 : [Nat8] = [0x00];
+          let OP_SEPARATOR_POS : [Nat8] = [0xFF, 0xFF, 0xFF, 0xFF];
+          ([0x02], Array.flatten([leaf_hash, KEY_VERSION_0, OP_SEPARATOR_POS]));
+        };
+        case (null) {
+          ([0x00], []);
+        };
+      };
```

Comment:
this is defined in https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki#common-signature-message-extension, added a comment

Reply:
As discussed offline, we don't use an annex, so `annex_present` is `0`, based on BIP341 which says:

> Additionally, future soft forks can make use of the currently unused annex in the witness

> What is the purpose of the annex? The annex is a reserved space for future extensions, such as indicating the validation costs of computationally expensive new opcodes in a way that is recognizable without knowing the scriptPubKey of the output being spent. Until the meaning of this field is defined by another softfork, users SHOULD NOT include annex in transactions, or it may lead to PERMANENT FUND LOSS.

With this, I'm resolving the comment.